js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,69 @@
+// describing spec (what's being tested)
+describe("enhancedDate", function () {
+
+	// Test each method within object
+
+	// setDate
+	describe("setDate", function () {
```

Comment:
One basic function I'd consider is simply checking the type returned by setDate:
`expect(enhancedDate.setDate()).to.be.an.instanceof(Date);`